### PR TITLE
ansible_test: Support receiving message to migrate guests

### PR DESCRIPTION
**Pull request by PaulYuuu**
Add a new scenario. After ansible-playbook runs, the test case will
launch an nc server and capture a specific message (regardless of
whether the message is written to the server through the playbook or
other forms). When the message is received, migrate guests and wait for
the execution of ansible-playbook to complete.
    
ID: 1912622
Depends on: #2591
Signed-off-by: Yihuang Yu <yihyu@redhat.com>

### CODE REVIEW

**Comment by PaulYuuu** [comment]
Hello @qcheng-redhat @luckyh , please help review this patch, and it depends on #2591. Thank you.

File: qemu/tests/cfg/ansible_test.cfg [hunk 1]
```diff
@@ -9,3 +9,12 @@
     ansible_ssh_extra_args = "-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
     # Here we can define an extra set of variables for the playbook with json format
     #ansible_extra_vars = '{"debug_msg": "Hello Ansible!", "force_handlers": true}'
+    variants guest_operation:
+        - @default:
+        - responsive_migration:
```
Comment (qcheng-redhat) [inline qemu/tests/cfg/ansible_test.cfg:14]: Need add toplevel_playbook = "test.yml" to start interactive_response on L1 guest, for me.

**Comment by qcheng-redhat** [comment]
In my environment, kar command execution failed on L1 guest, error:
['status: ERROR', 'fail reason: The remote server did not write the specified message within the expected time.', 'time: 1865.61s']
L2 guest starts up successfully, L1 gets message, but host doesn't get message, migration doesn't start.

**Comment by PaulYuuu** [comment]
@qcheng-redhat For internal information, please leave a comment to bugzilla.

File: qemu/tests/cfg/ansible_test.cfg (hunk 1 above)
Comment (PaulYuuu) [reply qemu/tests/cfg/ansible_test.cfg:14]: Internal only.

**Comment by qcheng-redhat** [comment]
python3 ConfigTest.py --testcase=ansible_test.responsive_migration --guestname=RHEL.8.4.0 --clone=no

 (1/1) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.ansible_test.responsive_migration: PASS (234.69 s)

File: qemu/tests/ansible_with_responsive_migration.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        callback_plugin=ansible_callback_plugin,
+        addl_opts=ansible_addl_opts
+    )
+
+    mq_publisher = message_queuing.MQPublisher(mq_listen_port)
+    try:
+        error_context.base_context('Confirm remote subscriber has accessed to '
+                                   'activate migrating guests.', logging.info)
+        try:
+            mq_publisher.confirm_access()(wait_response_timeout)
```
Comment (qcheng-redhat) [inline qemu/tests/ansible_with_responsive_migration.py:83]: mq_publisher.confirm_access()(wait_response_timeout)  >  duplicated bracket

**Comment by qcheng-redhat** [comment]
Another, when the first test case finished, nc connection will be terminated soon. When run the second test case with stress, I got: Ncat: Connection refused error.

File: qemu/tests/ansible_with_responsive_migration.py (hunk 2 above)
Comment (PaulYuuu) [reply qemu/tests/ansible_with_responsive_migration.py:83]: Thanks, I also noted it, will update in the next time.

**Comment by PaulYuuu** [comment]
> Another, when the first test case finished, nc connection will be terminated soon. When run the second test case with stress, I got: Ncat: Connection refused error.

This is expected. After the subscriber quits, the publisher will terminate soon, because the access process is one-time. To avoid migrate L1 multiple times, only one test case should be tested in L1.

**Comment by qcheng-redhat** [comment]
If only run one test case (the default one) on L1, I get pass:

python3 ConfigTest.py --testcase=ansible_test.responsive_migration --guestname=RHEL.8.4.0 --clone=no

(1/1) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.ansible_test.responsive_migration: PASS (227.35 s)

File: qemu/tests/ansible_test.py [hunk 3]
```diff
@@ -58,27 +57,22 @@ def run(test, params, env):
...
+    playbook_executor = PlaybookExecutor(
+        inventory="{},".format(",".join(guest_ip_list)),
+        site_yml=toplevel_playbook,
+        remote_user=guest_user,
+        extra_vars=json.dumps(extra_vars),
+        callback_plugin=ansible_callback_plugin,
+        addl_opts=ansible_addl_opts
+    )
+
+    playbook_executor.wait_for_completed(playbook_timeout, step_time)
```
Comment (qcheng-redhat) [inline qemu/tests/ansible_test.py:70]: Need handle playbook timeout error. Now just logs the error message.

Comment (PaulYuuu) [reply qemu/tests/ansible_test.py:70]: After the ansible-playbook terminated, only logs the error message, the full output will store into ansible_playbook.log 
```
playbook_executor.store_playbook_log(test_harness_log_dir, ansible_log)
```

Comment (PaulYuuu) [reply qemu/tests/ansible_test.py]: Updated, please help review again, thanks.

**Comment by qcheng-redhat** [comment]
Migration with stress test:

(1/1) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.ansible_test.responsive_migration: ERROR: ansible-playbook cannot complete all tasks within the expected time. (441.22 s)

The error due to product bug:
https://bugzilla.redhat.com/show_bug.cgi?id=1906893

**Comment by qcheng-redhat** [comment]
LGTM

**Pull request merged**